packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -759,49 +755,302 @@ class CupertinoActionSheetAction extends StatelessWidget {
+  _ActionSheetButtonBackgroundState createState() => _ActionSheetButtonBackgroundState();
 }
 
-class _CupertinoActionSheetCancelButtonState extends State<_CupertinoActionSheetCancelButton> {
+class _ActionSheetButtonBackgroundState extends State<_ActionSheetButtonBackground> {
   bool isBeingPressed = false;
 
   void _onTapDown(TapDownDetails event) {
     setState(() { isBeingPressed = true; });
+    widget.onStateChange(true);
   }
 
   void _onTapUp(TapUpDetails event) {
     setState(() { isBeingPressed = false; });
+    widget.onStateChange(false);
   }
 
   void _onTapCancel() {
     setState(() { isBeingPressed = false; });
+    widget.onStateChange(false);
   }
 
   @override
   Widget build(BuildContext context) {
-    final Color backgroundColor = isBeingPressed
-        ? _kActionSheetCancelPressedColor
+    late final Color backgroundColor;
+    BorderRadius? borderRadius;
+    if (!widget.isCancel) {
+      backgroundColor = isBeingPressed
+        ? _kPressedColor
+        : CupertinoDynamicColor.resolve(_kActionSheetBackgroundColor, context);
+    } else {
+      backgroundColor = isBeingPressed
+          ? _kActionSheetCancelPressedColor
         : CupertinoColors.secondarySystemGroupedBackground;
+      borderRadius = const BorderRadius.all(Radius.circular(_kCornerRadius));
+    }
     return GestureDetector(
       excludeFromSemantics: true,
       onTapDown: _onTapDown,
       onTapUp: _onTapUp,
       onTapCancel: _onTapCancel,
       child: Container(
         decoration: BoxDecoration(
-          color: CupertinoDynamicColor.resolve(backgroundColor, context),
-          borderRadius: const BorderRadius.all(Radius.circular(_kCornerRadius)),
+          color: backgroundColor,
+          borderRadius: borderRadius,
         ),
         child: widget.child,
+      )
+    );
+  }
+}
+
+// The divider of an action sheet.
+//
+// If the divider is not `hidden`, then it displays the `dividerColor`.
+// Otherwise it displays the background color. A divider is hidden when either
+// of its neighbor button is pressed.
+class _ActionSheetDivider extends StatelessWidget {
+  const _ActionSheetDivider({
+    super.key,
+    required this.dividerColor,
+    required this.hidden,
+  });
+
+  final Color dividerColor;
+  final bool hidden;
+
+  @override
+  Widget build(BuildContext context) {
+    final Color backgroundColor = CupertinoDynamicColor.resolve(_kActionSheetBackgroundColor, context);
+    return Container(
+      height: _kDividerThickness,
+      decoration: BoxDecoration(
+        color: hidden ? backgroundColor : dividerColor,
+      ),
+    );
+  }
+}
+
+// The keys used by the dividers in the action section of an action sheet.
+//
+// The buttons and the dividers are placed alternately, both keyed by indexes.
+// Dividers use this special class so that the two lists of indexes can be told
+// apart.
+class _DividerKey extends ValueKey<int> {
```

Comment:
I am curious why this is needed, are you seeing some state lost due if not using this?

Reply:
I wanted to add keys to each element of the list. Since there are two list of integers, I can't add, for example, `ValueKey<int>(1)` to both the second item and the first divider (I've tested that in this case Flutter will report duplicate key error.)

If you mean that we don't need keys, well, I'm not strong on it either. But I was thinking that adding keys to a list of items might improve the performance, or at least is a good practice. What do you think?

Reply:
key in a multichildrenderobject will only matter if you reorder or add or remove child in between update and you want to maintain state, otherwise it is the same as unkeyed, so I will probably not bother adding one for the divider

Reply:
Done, removed!